tools/setup_boost.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+set -e
```

Comment:
I think it would be helpful with a comment here, describing what this script does and what it's used for. I don't think it's obvious that it's used for CI